Spatial_Engine/Modify/RoundCoordinates.cs

```diff
@@ -80,11 +79,21 @@ public static IElement2D RoundCoordinates(this IElement2D element2d, int decimal
 
                 return element2d.ISetInternalElements2D(element2d.IInternalElements2D().Select(y => y.ISetOutlineElements1D(y.IOutlineElements1D().Select(x => x.ISetGeometry(Geometry.Modify.IRoundCoordinates(x.IGeometry().IProject(plane), decimalPlaces))).ToList())).ToList());
             }
+            else
+            {
+                IElement2D newElement2d = element2d.ISetOutlineElements1D(element2d.IOutlineElements1D().Select(x => x.ISetGeometry(Geometry.Modify.IRoundCoordinates(x.IGeometry(), decimalPlaces))).ToList());
+                
+                newElement2d.ISetInternalElements2D(newElement2d.IInternalElements2D().Select(y => y.ISetOutlineElements1D(y.IOutlineElements1D().Select(x => x.ISetGeometry(Geometry.Modify.IRoundCoordinates(x.IGeometry(), decimalPlaces))).ToList())).ToList());
 
-            Reflection.Compute.RecordWarning("Rounding the coordinates of a planar surface that is not aligned with the global coordinate system cannot be achieved without risk of losing planarity. No action has been taken.");
+                if (newElement2d.IElementCurves().IClusterCoplanar().Count == 1)
```

Comment:
This looks a bit like counting sheep by counting their legs and dividing by 4 🐑

Reply:
Agreed. What do you think of something like this?
```suggestion
                if (newElement2d.IElementCurves().SelectMany(c => c.SamplePoints(10)).ToList().IsCoplanar())
```

Reply:
Following off-air talk I'll just add an IsPlanar method for `IElements`.